src/main/java/org/candlepin/cache/ConsumerCertificateContentCache.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.cache;
+
+import org.candlepin.config.ConfigProperties;
+import org.candlepin.config.Configuration;
+import org.candlepin.config.ConfigurationException;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * A thread safe cache for consumer certificate content. Entries are evicted based on a time-to-live eviction
+ * policy.
+ */
+@Singleton
+public class ConsumerCertificateContentCache {
+    private Cache<Set<String>, ConsumerCertContent> cache;
```

Comment:
We should consider more specialized cache implementation like Caffeine. It [offers better performance](https://github.com/ben-manes/caffeine/wiki/Benchmarks) and is based on lessons learned from Guava's implementation.

Reply:
Oh wow! The performance difference is pretty incredible when comparing to Guava.  I would be on board with using Caffeine!

Reply:
Updated to use Caffeine